plugins/device/rest.go

```diff
@@ -148,6 +148,12 @@ func deviceEditHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	deviceName := r.URL.Query().Get("device")
+	if deviceName == "" {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Device name not provided in URL")
+		return
+	}
+
 	txn, err := transaction.NewTxnWithLocks(ctx, peerID)
```

Comment:
lock peer + device before editing it, so it won't be considered for brick/volume creation during device edit.

Reply:
lock peer+device similar to delete device